notes/stv/20200507-hadoop-k8s.txt

```diff
@@ -0,0 +1,422 @@
+# ----------------------------------------------------------
+
+
+
+# Get Pods
+# -------------
+kubectl get pods -n default
+NAME                            READY   STATUS    RESTARTS   AGE
+hadoop-hadoop-hdfs-dn-0         1/1     Running   2          128m
+hadoop-hadoop-hdfs-nn-0         1/1     Running   0          128m
+hadoop-hadoop-yarn-nm-0         1/1     Running   0          128m
+hadoop-hadoop-yarn-nm-1         1/1     Running   0          128m
+hadoop-hadoop-yarn-rm-0         1/1     Running   0          128m
+spark-master-controller-4jnbd   1/1     Running   0          9d
+spark-worker-controller-4jvh5   1/1     Running   0          9d
+spark-worker-controller-rw4pq   1/1     Running   0          9d
+
+
+# There are 5 Hadoop pods:
+1 data node
+3 name nodes
+1 resource manager node
+
+# I assume 1 datanode cause "replicas" param defaults to one?
+
+
+
+# Get Services
+# -------------
+kubectl get services     
+NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE
+hadoop-hadoop-hdfs-dn   ClusterIP   None             <none>        9000/TCP,50075/TCP           132m
+hadoop-hadoop-hdfs-nn   ClusterIP   None             <none>        9000/TCP,50070/TCP           132m
+hadoop-hadoop-yarn-nm   ClusterIP   None             <none>        8088/TCP,8082/TCP,8042/TCP   132m
+hadoop-hadoop-yarn-rm   ClusterIP   None             <none>        8088/TCP                     132m
+hadoop-hadoop-yarn-ui   ClusterIP   10.254.142.3     <none>        8088/TCP                     132m
+kubernetes              ClusterIP   10.254.0.1       <none>        443/TCP                      10d
+spark-master            ClusterIP   10.254.252.249   <none>        7077/TCP,8080/TCP            9d
+spark-master-headless   ClusterIP   None             <none>        <none>                       9d
+
+
+# Get Persistent Storage
+# ----------------------
+kubectl get pv --sort-by=.spec.capacity.storage
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                           STORAGECLASS   REASON   AGE
+pvc-e62eef27-6a06-447d-867f-13407c6e6fa7   50Gi       RWO            Delete           Bound    default/hadoop-hadoop-hdfs-nn   standard                139m
+pvc-b995c154-ddd0-4fdb-a48a-0a25b75a5f06   200Gi      RWO            Delete           Bound    default/hadoop-hadoop-hdfs-dn   standard                140m
+
+
+
+
+
+
+# ----------------------------------------------------------
+# Put some example files into HDFS
+# ----------------------------------------------------------
+# Exec into a NameNode pod, and run some HDFS commands
+#
+#
+
+kubectl exec -it -n default hadoop-hadoop-hdfs-nn-0 bash
+
+cd ~
+hdfs dfs -mkdir /hadoop/
+hdfs dfs -mkdir /hadoop/books/
+
+wget -O alice.txt https://www.gutenberg.org/files/11/11-0.txt
+wget -O holmes.txt https://www.gutenberg.org/files/1661/1661-0.txt
+wget -O frankenstein.txt https://www.gutenberg.org/files/84/84-0.txt
```

Comment:
We should be loading and testing with test data based on Gaia.
I've created issue #127 for this.